### PR TITLE
Time remaining section

### LINKED ISSUE #71: Time remaining - MAIN SITE and LIVE SITE
- [x] Create a component, TimeRemaining in shared-ui 
- [x] Develop it according to these mobile + desktop designs 
Screenshot 2022-11-16 at 6 57 36 PM
Screenshot 2022-11-16 at 6 57 58 PM

### PR DESCRIPTION
**Pull request by srazh**
Implemented a countdown to first day of Hackathon

Fixes #71 

Changelist:

- implemented TimeRemaining component in Live and Main site

Notes:
- will update once we have official date --right now its set for 02/03/2023

Tested:

- local host 8000

Screenshots & Screencasts:

Screen Shot 2022-11-21 at 3 49 03 PM
Screen Shot 2022-11-21 at 3 48 16 PM

### CODE REVIEW

File: shared-ui/components/time-remaining/TimeRemaining.styles.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.styles.tsx]: Can we remove all the random new lines heh

File: shared-ui/components/time-remaining/TimeRemaining.styles.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.styles.tsx:32]: here too

File: shared-ui/components/time-remaining/TimeRemaining.styles.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.styles.tsx]: here too

File: shared-ui/components/time-remaining/TimeRemaining.styles.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.styles.tsx]: here too

File: shared-ui/components/time-remaining/TimeRemaining.styles.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.styles.tsx]: here too

File: shared-ui/components/time-remaining/TimeRemaining.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.tsx]: Let's add a return type here like in the other files! It can just be React.FC

File: shared-ui/components/time-remaining/TimeRemaining.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.tsx]: These are all pretty much the same thing except for the unit of time that's different. To reduce this code, let's create an array of TimeUnit which would look something like {text: seconds, label: 'Seconds'}. And then we can map over your array instead of having 

```
 <StyledTimeRemainingText>{minutes}  :</StyledTimeRemainingText>
                    <StyledTimeRemainingLabel>Minutes</StyledTimeRemainingLabel>
                </div>
```

this chunk 4 times

File: shared-ui/components/time-remaining/TimeRemaining.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.tsx]: Do you have the prettier and eslint extensions? If not could you please install them in vscode? And then run the prettier formatter on all ur file heh some of the formatting looks a bit funky

File: shared-ui/components/time-remaining/TimeRemaining.tsx
Comment (jzhzhang66) [inline shared-ui/components/time-remaining/TimeRemaining.tsx]: For this button can we change the functionality to be copying to cliipboard? (like how it was on the old site)? Since signing yourself up for a mailing list is pretty different than sharing about the even with friends LOL

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
LOOKS GREAT SARAH!!! KILLED IT!! 

Just a few small comments and also, I just asked David + Sreya if we should wait until a date is finalized for this to be merged in or if we should just use 2/3!

**Comment by jzhzhang66** [comment]
Also, could you please resolve the conflicts in the colors.ts?

**Review by jzhzhang66 — CHANGES_REQUESTED** [review]
Could we also add a bit more bottom padding to this? Looks a bit too close to the footer in comparison to how close other sections are to each other
Screenshot 2022-11-21 at 9 17 05 PM

**Comment by deanframe** [comment]
could we add some top padding to the counter? looks a little squished to the top of the page

**Review by deanframe — COMMENTED** [review]
Looks pretty good! +1 to all of judy's requested changes, will approve after.

FYI theres a date library called `moment` that a lot of people prefer using over vanilla js dates - maybe it could have been helpful here (not really sure tbh, dont know a whole lot about it). Please don't rewrite this hahaha unless you get bored over break, just wanted to put it on your radar

**Review by jzhzhang66 — APPROVED** [review]

**Review by deanframe — APPROVED** [review]
lgtm!

**Pull request merged**
